src/shaders/patternshader.cpp

```diff
@@ -0,0 +1,35 @@
+#include "shaders/patternshader.h"
+
+using namespace mixxx;
+
+void PatternShader::init() {
+    QString vertexShaderCode = QStringLiteral(R"--(
+uniform highp mat4 matrix;
+attribute highp vec4 position;
```

Comment:
yes, because it is being multiplied with a mat4... which indeed could also be a mat2. i will do some additional cleanup of the shaders, while at it.

Reply:
of course the day we want waveforms projected on rotating 3d cubes will will have to revert this ;-)

Reply:
i will do a bit of type cleanup though

Reply:
> there is no advantage in doing the conversion from vec2 to vec4 manually, and doing it like this keeps the shader more flexible (usuable for 3d, which is of course not our usecase, but since we get it for free, why cripple the shader?)

Sure, makes sense. the vec3 is probably padded to 4 anyways. This just needs a comment then.